trying to add heating coils in GregTech causes a funny gregtech.material.null issue [Bug reported in discord server transferred to github]

### GregTech CEu Version

GregTech CEu V2.7,3-beta

### Modpack Used

First spotted while working on CBT: New Horizons 1.0,4

### Addons Installed

Addons were installed but are irrelevant to the bug.

### Environment

In-game

### New Worlds

Unsure

### Cross-Mod Interaction

Yes

### Expected Behavior

The coil should be present in the game

### Actual Behavior

Uhhhhhhhhhhhhhhhhhhhhhhh

![image](https://github.com/GregTechCEu/GregTech/assets/141803157/785d5a7e-7be9-4244-b4e5-00179655cc2f)
![image](https://github.com/GregTechCEu/GregTech/assets/141803157/5c811ed7-da4c-4be5-b92e-d552fa549ddc)
![image](https://github.com/GregTechCEu/GregTech/assets/141803157/0074e64c-78fe-4b13-8cd5-050193f52128)
![image](https://github.com/GregTechCEu/GregTech/assets/141803157/d5a458b8-fecd-45bb-a694-86e49d6da34e)

so basically i was trying to add this block using crafttweaker

i tried multiple different iterations:
HeatingCoils.add(<blockstate:gregtech:meta_block_compressed_937:variant=gregtech_schrabidium>, "schrabidium", 24000, 2, 4, 3, <material:schrabidium>);
HeatingCoils.add(<blockstate:gregtech:meta_block_compressed_937:15:variant=gregtech_schrabidium>, "schrabidium", 24000, 2, 4, 3, <material:schrabidium>);

Without the variant thing it straight up crashes.


### Steps to Reproduce

1. add any metablocks from GTCEu as a base for the heating coil block using crafttweaker
2. double check for the variant
3. launch game and if all goes right, it should be registered.. 
4. thats not the end, if the game crashes, fix the code. if game doesn't crash, look for the multiblock recipe and you would see a gregtech,material.null block in place 

### Additional Information

Details of the bug:
This bug has 2 mods involved:
GTCEu (main bug)
CraftTweaker (for easy reproduction)

This bug was originally reported in the discord server, but transferred here so there won't be 2 different reports of the same bug. 

Could not resolve line [INITIALIZATION][CLIENT][ERROR] periodic table\new_casings.zs:11: Could not resolve <gregtech : schrabidium> that happens with the crash

Once the fix is applied, this bug report can be closed.

--- a/src/main/java/gregtech/common/blocks/properties/PropertyMaterial.java
+++ b/src/main/java/gregtech/common/blocks/properties/PropertyMaterial.java
@@ -37,8 +37,8 @@ public class PropertyMaterial extends PropertyHelper<Material> {
     @Nonnull
     @Override
     public Optional<Material> parseValue(@Nonnull String value) {
-        int index = value.indexOf(':');
-        String materialName = index < 0 ? value : value.substring(0, index) + '_' + value.substring(index + 1);
+        int index = value.indexOf("__");
+        String materialName = index < 0 ? value : value.substring(0, index) + ':' + value.substring(index + 2);
         Material material = GregTechAPI.materialManager.getMaterial(materialName);
         if (material != null && this.allowedValues.contains(material)) {
             return Optional.of(material);
@@ -49,7 +49,8 @@ public class PropertyMaterial extends PropertyHelper<Material> {
     @Nonnull
     @Override
     public String getName(@Nonnull Material material) {
-        return material.getModid() + '_' + material.getName();
+        // Use double underscore to prevent ${modid}_${material_name} being ambiguous with ${material_name} when parsing
+        return material.getModid() + "__" + material.getName();
     }
 
     @Override

